Food_Tracker_App/app/src/main/java/com/example/foodtracker/ui/recipes/AddRecipeActivity.java

```diff
@@ -101,24 +161,61 @@ public void initializeRecipe(Recipe recipe) {
         }
     }
 
+    /**
+     * Populates the dialog fields from a {@link Recipe} instance
+     *
+     * @param recipe to initialize form with
+     */
+    public void initializeEditRecipe(Recipe recipe) {
+        if (!recipe.getImage().isEmpty()) {
+            imageURI = Uri.parse(recipe.getImage());
+            recipeImage.setImageURI(imageURI);
+        }
+
+        titleField.setText(recipe.getTitle());
+        timeField.setText(String.valueOf(recipe.getPrepTime()));
+        servingsField.setText(String.valueOf(recipe.getServings()));
+        commentsField.setText(recipe.getComment());
+
+        ingredientList = recipe.getIngredients();
+        adapter = new CustomList(this, ingredientList);
+        ingredientsListField.setAdapter(adapter);
+    }
+
     /**
      * Retrieves categories from firestore and populates a string array with the content
      */
-    private void getCategories() {
+    private void getCategories(@Nullable Recipe recipe) {
         categoryCollection.getAll(list -> {
-            for (Category category : list) {
+            for (com.example.foodtracker.model.recipe.Category category : list) {
                 categories.add(category.getName());
                 categoryAdapter.notifyDataSetChanged();
             }
+            if (recipe != null) {
+                categoryField.setSelection(categoryAdapter.getPosition(recipe.getCategory()));
+            }
         });
     }
 
     private void addImageFromGallery() {
         imageGalleryResultHandler.launch("image/*");
     }
 
-    private Recipe createRecipe() {
+    /**
+     * Creating a recipe with the details in the fields
+     * @return {@link Recipe} recipe the created recipe
+     */
+    private Boolean createRecipe(Recipe recipe) {
```

Comment:
doesn't really create anything, maybe rename this to `setRecipeFields`? and change the javadoc